osu.Game/Overlays/Options/General/LoginOptions.cs

```diff
@@ -76,41 +74,77 @@ public void APIStateChanged(APIAccess api, APIState state)
+            }
+
+            private void performLogin()
             {
+                if (!string.IsNullOrEmpty(username.Text) && !string.IsNullOrEmpty(password.Text))
+                    api.Login(username.Text, password.Text);
+            }
+
+            [BackgroundDependencyLoader(permitNulls: true)]
+            private void load(APIAccess api, OsuConfigManager config)
+            {
+                this.api = api;
                 Direction = FlowDirection.VerticalOnly;
                 AutoSizeAxes = Axes.Y;
                 RelativeSizeAxes = Axes.X;
                 Spacing = new Vector2(0, 5);
-                // TODO: Wire things up
                 Children = new Drawable[]
                 {
                     new SpriteText { Text = "Username" },
-                    username = new TextBox { Height = 20, RelativeSizeAxes = Axes.X, Text = api?.Username ?? string.Empty },
+                    username = new TextBox
+                    {
+                        Height = 20,
+                        RelativeSizeAxes = Axes.X,
+                        Text = api?.Username ?? string.Empty
+                    },
                     new SpriteText { Text = "Password" },
-                    password = new PasswordTextBox { Height = 20, RelativeSizeAxes = Axes.X },
+                    password = new PasswordTextBox
+                    {
+                        Height = 20,
+                        RelativeSizeAxes = Axes.X
+                    },
+                    saveUsername = new CheckBoxOption
+                    {
+                        LabelText = "Remember Username",
+                        Bindable = config.GetBindable<bool>(OsuConfig.SaveUsername),
+                    },
+                    savePassword = new CheckBoxOption
+                    {
+                        LabelText = "Remember Password",
+                        Bindable = config.GetBindable<bool>(OsuConfig.SavePassword),
+                    },
                     new OsuButton
                     {
                         RelativeSizeAxes = Axes.X,
                         Text = "Log in",
                         Action = performLogin
+                    },
+                    new OsuButton
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        Text = "Register",
+                        //Action = registerLink
                     }
-                };
-            }
-
-            private void performLogin()
-            {
-                if (!string.IsNullOrEmpty(username.Text) && !string.IsNullOrEmpty(password.Text))
-                    api.Login(username.Text, password.Text);
-            }
-
-            [BackgroundDependencyLoader(permitNulls: true)]
-            private void load(APIAccess api)
-            {
-                this.api = api;
+                };
+                config.GetBindable<bool>(OsuConfig.SavePassword).ValueChanged += delegate { eventPassword(); };

```

Comment:
this logic should be moved to `OsuConfigManager`, since it's describing the logic between two config bindables.